src/scripts/main.js

```diff
@@ -1,3 +1,54 @@
 'use strict';
 
-// write your code here
+const logo = document.querySelector('.logo');
+
+function waitForClick(element) {
+  return new Promise((resolve) => {
+    element.addEventListener('click', () => resolve('Element was clicked'), {
+      once: true,
+    });
+  });
+}
+
+function successMessage() {
+  const div = document.createElement('div');
+
+  div.innerText = 'Promise was resolved!';
+  div.className = 'message';
+
+  return div;
+}
+
+function errorMesssage(err) {
+  const div = document.createElement('div');
+
+  div.innerText = 'Promise was rejected!';
+  div.className = 'message error-message';
+  Error(err);
```

Comment:
The `Error` object is created but not used properly. If you intend to log or throw the error, consider using `console.error(err)` or `throw new Error(err)` instead.